Modules/IO/ImageBase/include/itkImageSeriesWriter.hxx

```diff
@@ -140,7 +140,10 @@ ImageSeriesWriter<TInputImage, TOutputImage>::GenerateNumericFileNames()
 
   for (unsigned int slice = 0; slice < numberOfFiles; ++slice)
   {
+    ITK_GCC_PRAGMA_PUSH
+    ITK_GCC_SUPPRESS_Wformat_nonliteral
     snprintf(fileName, IOCommon::ITK_MAXPATHLEN + 1, m_SeriesFormat.c_str(), fileNumber);
+    ITK_GCC_PRAGMA_POP
```

Comment:
I tried, but could not think of / find a good/easy way to pre-flight a user-provided format string.

I'm no C++20 expert, but quickly reading that link says "the format string must be passed in as a compile-time constant, and std::format does a compile-time check on it" so that's basically the same as `snprintf`. The whole problem here is that it's user-provided, not compile-time constant.

Reply:
@N-Dekker Do you have a suggestion for a more secure fix that would be backward compatible, and not require a huge code rewrite?

Your suggestion to make the code more secure, but I think it should be a separate issue. 

 See:  https://github.com/InsightSoftwareConsortium/ITK/issues/4621

Reply:
Opening that ticket for future work is a good idea. This PR will at least get cdash greener and we can then ensure no future such warnings are introduced.